### ISSUE TITLE
spp fails processing one pseudo rep

### ISSUE THREAD

**Issue by gevirl** [ENCODE-DCC/chip-seq-pipeline2#126]
**Describe the bug**
spp fails in step call-call_peak_pr2 for shard-1 , but has not problem with shard-0

**OS/Platform**
-Centos 6.10  and UGE 
- Conda version: 4.7.12
- Pipeline version: v1.3.2
- Caper version:0.6.3

**Caper configuration file**
backend=sge
sge-pe=serial

# DO NOT use /tmp here
# Caper stores all important temp files and cached big data files here
tmp-dir=/net/waterston/vol9

**Input JSON file**
{"chip.title":"dmd-8_OP775_L4larva_1_1","chip.description":"gevirl","chip.always_use_pooled_ctl":false,"chip.true_rep_only":false,"chip.enable_count_signal_track":true,"chip.aligner":"bwa","chip.use_bwa_mem_for_pe":true,"chip.align_only":false,"chip.genome_tsv":"/net/waterston/vol9/WS245/WS245.tsv","chip.peak_caller":"spp","chip.pipeline_type":"tf","chip.idr_thresh":0.05,"chip.call_peak_mem_mb":16000,"chip.align_mem_mb":24000,"chip.fastqs_rep1_R1":["/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/DMDIP1_018_175_S73_L004_R1_001.fastq.gz"],"chip.fastqs_rep1_R2":["/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/DMDIP1_018_175_S73_L004_R2_001.fastq.gz"],"chip.fastqs_rep2_R1":["/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/DMDIP2_006_187_S74_L004_R1_001.fastq.gz"],"chip.fastqs_rep2_R2":["/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/DMDIP2_006_187_S74_L004_R2_001.fastq.gz"],"chip.paired_ends":[true,true],"chip.ctl_fastqs_rep1_R1":["/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/DMDinp_089_104_S75_L004_R1_001.fastq.gz"],"chip.ctl_fastqs_rep1_R2":["/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/DMDinp_089_104_S75_L004_R2_001.fastq.gz"],"chip.ctl_paired_ends":[true]}

**Error log**
Caper automatically runs a troubleshooter for failed workflows. If it doesn't then get a `WORKFLOW_ID` of your failed workflow with `caper list` or directly use a `metadata.json` file on Caper's output directory.
```
$ caper debug [WORKFLOW_ID_OR_METADATA_JSON_FILE]
caper debug metadata.json
[Caper] troubleshooting 103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3 ...
Found failures:
[
    {
        "causedBy": [
            {
                "message": "Job chip.call_peak_pr2:1:2 exited with return code 1 which has not been declared as a valid return code. See 'continueOnReturnCode' runtime attribute for more details.",
                "causedBy": []
            }
        ],
        "message": "Workflow failed"
    }
]

chip.call_peak_pr2 RetryableFailure. SHARD_IDX=1, RC=1, JOB_ID=151133162, RUN_START=2020-02-07T17:32:28.216Z, RUN_END=2020-02-07T18:05:01.934Z, STDOUT=/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/execution/stdout, STDERR=/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/execution/stderr
STDERR_CONTENTS=
Traceback (most recent call last):
  File "/net/waterston/vol2/home/waterston-jboss/miniconda3/envs/encode-chip-seq-pipeline/bin/encode_task_spp.py", line 103, in <module>
    main()
  File "/net/waterston/vol2/home/waterston-jboss/miniconda3/envs/encode-chip-seq-pipeline/bin/encode_task_spp.py", line 91, in main
    args.fraglen, args.cap_num_peak, args.nth, args.out_dir)
  File "/net/waterston/vol2/home/waterston-jboss/miniconda3/envs/encode-chip-seq-pipeline/bin/encode_task_spp.py", line 66, in spp
    run_shell_cmd(cmd0)
  File "/net/waterston/vol2/home/waterston-jboss/miniconda3/envs/encode-chip-seq-pipeline/bin/encode_lib_common.py", line 319, in run_shell_cmd
    raise Exception(err_str)
Exception: PID=4156, PGID=4156, RC=1
STDERR=Loading required package: Rcpp
Error in lwcc(x, y, s, e, return.peaks = return.peaks, bg.x = bg.x, bg.y = bg.y,  : 
  INTEGER() can only be applied to a 'integer', not a 'NULL'
Calls: find.binding.positions ... FUN -> window.chr.call.mirror.binding -> method -> lwcc
Execution halted
STDOUT=################
ChIP data: /net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/inputs/1771568480/DMDIP2_006_187_S74_L004_R1_001.merged.nodup.pr2.tagAlign.gz 
Control data: /net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/inputs/2045518733/ctl_for_rep2.tagAlign.gz 
strandshift(min): -500 
strandshift(step): 5 
strandshift(max) 1500 
user-defined peak shift 150 
exclusion(min): 10 
exclusion(max): NaN 
num parallel nodes: NA 
FDR threshold: 0.01 
NumPeaks Threshold: 3e+05 
Output Directory: /net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/execution 
narrowPeak output file name: NA 
regionPeak output file name: DMDIP2_006_187_S74_L004_R1_001.merged.nodup.pr2_x_ctl_for_rep2.300K.regionPeak.gz.tmp 
Rdata filename: NA 
plot pdf filename: NA 
result filename: NA 
Overwrite files?: TRUE

Decompressing ChIP file
Decompressing control file
Reading ChIP tagAlign/BAM file /net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/inputs/1771568480/DMDIP2_006_187_S74_L004_R1_001.merged.nodup.pr2.tagAlign.gz 
opened /net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/tmp.f62e18c0/RtmptCaU3P/DMDIP2_006_187_S74_L004_R1_001.merged.nodup.pr2.tagAlign103e4a4390ed
done. read 8804966 fragments
ChIP data read length 101 
[1] TRUE
Reading Control tagAlign/BAM file /net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/inputs/2045518733/ctl_for_rep2.tagAlign.gz 
opened /net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/tmp.f62e18c0/RtmptCaU3P/ctl_for_rep2.tagAlign103e26419335
done. read 18000300 fragments
Control data read length 101 
Calculating peak characteristics
Minimum cross-correlation value 0.770026 
Minimum cross-correlation shift 1500 
Top 3 cross-correlation values 0.784787400538734 
Top 3 estimates for fragment length 150 
Window half size 230 
Phantom peak location 105 
Phantom peak Correlation 0.7851819 
Normalized Strand cross-correlation coefficient (NSC) 1.01917 
Relative Strand cross-correlation Coefficient (RSC) 0.9739724 
Phantom Peak Quality Tag 0 
Removing read stacks
Finding peaks
finding background exclusion regions ... done
determining peaks on provided 1 control datasets:
using reversed signal for FDR calculations
bg.weight= 2.055662  processing X in 1 steps [.] done ( 27093 positions)
processing V in 1 steps [
ln: accessing `/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/execution/*[!.][!b][!f][!i][!l][!t].regionPeak.gz': No such file or directory
ln: accessing `/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/execution/*.bfilt.regionPeak.gz': No such file or directory
ln: accessing `/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/execution/*.bfilt.regionPeak.bb': No such file or directory
ln: accessing `/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/execution/*.bfilt.regionPeak.hammock.gz*': No such file or directory
mkdir: cannot create directory `/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/execution/glob-c71d4cca8627aab27dc3503b7db7a39d': File exists
ln: accessing `/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/execution/*.bfilt.regionPeak.hammock.gz*': No such file or directory
ln: accessing `/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/execution/*.frip.qc': No such file or directory
ln: accessing `/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/execution/*.peak_region_size.qc': No such file or directory
ln: accessing `/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/execution/*.peak_region_size.png': No such file or directory
ln: accessing `/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/execution/*.num_peak.qc': No such file or directory


chip.call_peak_pr2 Failed. SHARD_IDX=1, RC=1, JOB_ID=151134594, RUN_START=2020-02-07T18:05:04.213Z, RUN_END=2020-02-07T18:38:03.771Z, STDOUT=/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/attempt-2/execution/stdout, STDERR=/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/attempt-2/execution/stderr
STDERR_CONTENTS=
Traceback (most recent call last):
  File "/net/waterston/vol2/home/waterston-jboss/miniconda3/envs/encode-chip-seq-pipeline/bin/encode_task_spp.py", line 103, in <module>
    main()
  File "/net/waterston/vol2/home/waterston-jboss/miniconda3/envs/encode-chip-seq-pipeline/bin/encode_task_spp.py", line 91, in main
    args.fraglen, args.cap_num_peak, args.nth, args.out_dir)
  File "/net/waterston/vol2/home/waterston-jboss/miniconda3/envs/encode-chip-seq-pipeline/bin/encode_task_spp.py", line 66, in spp
    run_shell_cmd(cmd0)
  File "/net/waterston/vol2/home/waterston-jboss/miniconda3/envs/encode-chip-seq-pipeline/bin/encode_lib_common.py", line 319, in run_shell_cmd
    raise Exception(err_str)
Exception: PID=12235, PGID=12235, RC=1
STDERR=Loading required package: Rcpp
Error in lwcc(x, y, s, e, return.peaks = return.peaks, bg.x = bg.x, bg.y = bg.y,  : 
  INTEGER() can only be applied to a 'integer', not a 'character'
Calls: find.binding.positions ... FUN -> window.chr.call.mirror.binding -> method -> lwcc
Execution halted
STDOUT=################
ChIP data: /net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/attempt-2/inputs/1771568480/DMDIP2_006_187_S74_L004_R1_001.merged.nodup.pr2.tagAlign.gz 
Control data: /net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/attempt-2/inputs/2045518733/ctl_for_rep2.tagAlign.gz 
strandshift(min): -500 
strandshift(step): 5 
strandshift(max) 1500 
user-defined peak shift 150 
exclusion(min): 10 
exclusion(max): NaN 
num parallel nodes: NA 
FDR threshold: 0.01 
NumPeaks Threshold: 3e+05 
Output Directory: /net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/attempt-2/execution 
narrowPeak output file name: NA 
regionPeak output file name: DMDIP2_006_187_S74_L004_R1_001.merged.nodup.pr2_x_ctl_for_rep2.300K.regionPeak.gz.tmp 
Rdata filename: NA 
plot pdf filename: NA 
result filename: NA 
Overwrite files?: TRUE

Decompressing ChIP file
Decompressing control file
Reading ChIP tagAlign/BAM file /net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/attempt-2/inputs/1771568480/DMDIP2_006_187_S74_L004_R1_001.merged.nodup.pr2.tagAlign.gz 
opened /net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/attempt-2/tmp.cacea48b/RtmpYgK4Lo/DMDIP2_006_187_S74_L004_R1_001.merged.nodup.pr2.tagAlign2fcd5e172b85
done. read 8804966 fragments
ChIP data read length 101 
[1] TRUE
Reading Control tagAlign/BAM file /net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/attempt-2/inputs/2045518733/ctl_for_rep2.tagAlign.gz 
opened /net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/attempt-2/tmp.cacea48b/RtmpYgK4Lo/ctl_for_rep2.tagAlign2fcd5bfa31af
done. read 18000300 fragments
Control data read length 101 
Calculating peak characteristics
Minimum cross-correlation value 0.770026 
Minimum cross-correlation shift 1500 
Top 3 cross-correlation values 0.784787400538734 
Top 3 estimates for fragment length 150 
Window half size 230 
Phantom peak location 105 
Phantom peak Correlation 0.7851819 
Normalized Strand cross-correlation coefficient (NSC) 1.01917 
Relative Strand cross-correlation Coefficient (RSC) 0.9739724 
Phantom Peak Quality Tag 0 
Removing read stacks
Finding peaks
finding background exclusion regions ... done
determining peaks on provided 1 control datasets:
using reversed signal for FDR calculations
bg.weight= 2.055662  processing X in 1 steps [.] done ( 27093 positions)
processing V in 1 steps [
ln: accessing `/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/attempt-2/execution/*[!.][!b][!f][!i][!l][!t].regionPeak.gz': No such file or directory
ln: accessing `/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/attempt-2/execution/*.bfilt.regionPeak.gz': No such file or directory
ln: accessing `/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/attempt-2/execution/*.bfilt.regionPeak.bb': No such file or directory
ln: accessing `/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/attempt-2/execution/*.bfilt.regionPeak.hammock.gz*': No such file or directory
mkdir: cannot create directory `/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/attempt-2/execution/glob-c71d4cca8627aab27dc3503b7db7a39d': File exists
ln: accessing `/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/attempt-2/execution/*.bfilt.regionPeak.hammock.gz*': No such file or directory
ln: accessing `/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/attempt-2/execution/*.frip.qc': No such file or directory
ln: accessing `/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/attempt-2/execution/*.peak_region_size.qc': No such file or directory
ln: accessing `/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/attempt-2/execution/*.peak_region_size.png': No such file or directory
ln: accessing `/net/waterston/vol9/ChipSeqPipeline/dmd-8_OP775_L4larva_1/chip/103b5ab5-d9d5-41ca-8b00-1e8fc71ef8b3/call-call_peak_pr2/shard-1/attempt-2/execution/*.num_peak.qc': No such file or directory

**Comment by leepc12** [comment]
This is a known issue of R-spp 1.16. Please use the latest pipeline v1.3.6, which is currently on the master branch. 
https://github.com/ENCODE-DCC/chip-seq-pipeline2/releases/tag/v1.3.6

Re-install your Conda env before running pipelines.
```
$ bash scripts/uninstall_conda_env.sh
$ bash scripts/install_conda_env.sh
```

**Comment by leepc12** [comment]
This should be fixed in the latest pipeline (v1.3.6). Closing this.
